test/trace-test.js

```diff
@@ -0,0 +1,135 @@
+    var opts = {expiresIn: 50, sink: {type: 'log'}};
+
+    // remove from nothing, then add, and again. should end with one tracer
+    store.remove(config, opts, function onRemove(err, tracer) {
+        assert.false(err, 'no error, idempotent remove');
+        assert.false(tracer, 'no tracer, idempotent remove');
+    });
+
+    var theTracer = store.add(config, opts, function onAdd(err, tracer) {
+        assert.false(err, 'no error, add good');
+        assert.true(tracer, 'tracer added');
+    });
+    assert.true(theTracer, 'tracer added, double checking');
+
+    store.add(config, opts, function onAdd(err, tracer) {
+        assert.false(err, 'no error, idempotent add');
+        assert.true(tracer, 'tracer already added');
+        assert.equal(tracer, theTracer, 'tracer returned is same');
+    });
+
+    // remove twice, checking both times
+    store.remove(config, opts, function onRemove(err, tracer) {
+        assert.false(err, 'no error, idempotent remove');
+        assert.true(tracer, 'tracer removed');
+        assert.equal(tracer, theTracer, 'tracer returned is same');
+    });
+
+    store.remove(config, opts, function onRemove(err, tracer) {
+        assert.false(err, 'no error, idempotent remove');
+        assert.false(tracer, 'tracer already removed');
+    });
+});
+
+test('trace events get traced', function t(assert) {
+    var ringpop = new rpMock();
+    var store = new TracerStore(ringpop);
+    var config = core.resolveEventConfig(ringpop, 'membership.update');
+    var opts = {expiresIn: 50, sink: {type: 'log'}};
+    var tracer = store.add(config, opts);
+
+    assert.plan(5);
+    tracer.on('connectSink', function onTracer() {
+        // now test events on this thing
+        assert.equal(ringpop.journal.length, 0, 'journal starts empty');
+        ringpop.membership.emit('updated', 'foo');
+        assert.equal(ringpop.journal.length, 1, 'journal has 1 entry');
+        assert.equal(ringpop.journal[0], 'foo', 'journal added foo');
+
+        ringpop.membership.emit('updated', 'bar');
+        assert.equal(ringpop.journal.length, 2, 'journal has 2 entries');
+        assert.equal(ringpop.journal[1], 'bar', 'journal added bar');
+    });
+});
+
+test('trace events can renew, and expire', function t(assert) {
+    var ringpop = new rpMock();
+    var store = new TracerStore(ringpop);
+    var config = core.resolveEventConfig(ringpop, 'membership.update');
+    var opts = {expiresIn: 50, sink: {type: 'log'}};
+    var tracer = store.add(config, opts);
+
+    assert.plan(3);
+
+    // nextTick okay since log "connects" instantly
+    process.nextTick(function onTracer() {
+        // now test events on this thing
+        ringpop.membership.emit('updated', 'foo');
+        assert.equal(ringpop.journal.length, 1, 'journal has 1 entry');
+
+        // renew after 25, run @ 60, confirm event; this is racey...
+        setTimeout(store.add.bind(store, config, opts), 25);
```

Comment:
Gladly. I hadn't seen time-mock (I looked briefly at stealing the time mock from sinon), and ultimately thought that these tiny timevals wouldn't be so bad, but I can easily rework to use the mock.

(Just checked, they add 140ms total time to the tests as-is. I still feel like mock time is better here for reliable timing, instead of the risk of system delays screwing up tests. But just thought I'd share.)